extensions/Worker.Extensions.Tables/src/TableConverter.cs

```diff
@@ -0,0 +1,175 @@
+        {
+            _logger = logger ?? throw new ArgumentNullException(nameof(logger));
+            _tableOptions = tableOptions ?? throw new ArgumentNullException(nameof(tableOptions));
+        }
+
+        public async ValueTask<ConversionResult> ConvertAsync(ConverterContext context)
+        {
+            return context?.Source switch
+            {
+                ModelBindingData binding => await ConvertFromBindingDataAsync(context, binding),
+                _ => ConversionResult.Unhandled()
+            };
+        }
+
+        public virtual async ValueTask<ConversionResult> ConvertFromBindingDataAsync(ConverterContext context, ModelBindingData modelBindingData)
+        {
+            if (!IsTableExtension(modelBindingData))
+            {
+                return ConversionResult.Unhandled();
+            }
+            try
+            {
+                Dictionary<string, object> content = GetBindingDataContent(modelBindingData);
+                var result = await ConvertModelBindingDataAsync(content, context.TargetType, modelBindingData);
+
+                if (result is not null)
+                {
+                    return ConversionResult.Success(result);
+                }
+            }
+            catch (Exception ex)
+            {
+                return ConversionResult.Failed(ex);
+            }
+
+            return ConversionResult.Unhandled();
+        }
+
+        internal bool IsTableExtension(ModelBindingData bindingData)
+        {
+            if (bindingData?.Source is not Constants.TablesExtensionName)
+            {
+                _logger.LogTrace("Source '{source}' is not supported by {converter}", bindingData?.Source, nameof(TableConverter));
+                return false;
+            }
+
+            return true;
+        }
+
+        public virtual async Task<object?> ConvertModelBindingDataAsync(IDictionary<string, object> content, Type targetType, ModelBindingData bindingData)
+        {
+            content.TryGetValue(Constants.Connection, out var connection);
+            content.TryGetValue(Constants.TableName, out var tableName);
+            content.TryGetValue(Constants.PartitionKey, out var partitionKey);
+            content.TryGetValue(Constants.RowKey, out var rowKey);
+            content.TryGetValue(Constants.Take, out var take);
+            content.TryGetValue(Constants.Filter, out var filter);
+
+            if (string.IsNullOrEmpty(tableName?.ToString()))
+            {
+                throw new ArgumentNullException("'TableName' cannot be null or empty");
+            }
+
+            return await ToTargetTypeAsync(targetType, connection?.ToString() ?? null, tableName!.ToString(), partitionKey?.ToString() ?? null, rowKey?.ToString() ?? null, Convert.ToInt32(take?.ToString()), filter?.ToString() ?? null);
+        }
+
+        internal Dictionary<string, object> GetBindingDataContent(ModelBindingData bindingData)
```

Comment:
I think all methods should be private here, only ConvertAsync (as the access point of the function) should be public